bin/prover-client/src/ckp_runner.rs

```diff
@@ -0,0 +1,151 @@
+use std::sync::Arc;
+
+use jsonrpsee::{
+    core::{client::ClientT, params::ArrayParams},
+    http_client::HttpClient,
+    rpc_params,
+};
+use strata_rpc_types::HexBytes;
+use tokio::time::{interval, Duration};
+use tracing::{error, info, warn};
+use uuid::Uuid;
+
+use crate::{
+    config::CHECKPOINT_POLL_INTERVAL,
+    dispatcher::TaskDispatcher,
+    primitives::tasks_scheduler::ProvingTaskStatus,
+    proving_ops::checkpoint_ops::{CheckpointOperations, CheckpointOpsParam},
+    task::TaskTracker,
+};
+
+/// Continuously checks for the latest checkpoint index.
+/// Dispatches tasks when a new checkpoint is detected and discards the current proving tasks.
+pub async fn start_checkpoints_task(
+    sequencer_client: HttpClient,
+    ckp_task_dispatcher: TaskDispatcher<CheckpointOperations>,
+    task_tracker: Arc<TaskTracker>,
+) {
+    info!("Checkpoint runner started");
+    let mut to_fetch_idx = 0;
+    let mut ticker = interval(Duration::from_secs(CHECKPOINT_POLL_INTERVAL));
+    let mut current_task_id = Uuid::default();
+
+    loop {
+        ticker.tick().await;
+
+        let new_checkpoint_idx =
+            fetch_latest_checkpoint_index(&sequencer_client, to_fetch_idx).await;
+
+        if new_checkpoint_idx >= to_fetch_idx {
+            match handle_new_checkpoint(new_checkpoint_idx, &ckp_task_dispatcher, &task_tracker)
+                .await
+            {
+                Ok(task_id) => {
+                    current_task_id = task_id;
+                    to_fetch_idx += 1;
```

Comment:
This should be `to_fetch_idx = new_checkpoint_idx + 1`. Or else, if initially to_fetch_idx is 0 and new_checkpoint_idx is say 10, this processes the checkpoint twice.